src/index.ts

```diff
@@ -1,25 +1,110 @@
-import { IfoodClient } from './ifood/index'
-
-import { Merchant } from './ifood/types/merchant'
-
-
-const main = async () => {
-  const ifoodClient = new IfoodClient()
-  const merchantsIds = (await ifoodClient.getMerchants()) as Merchant[]
-  const merchants = []
-  for (let index = 0; index < merchantsIds.length; index++) {
-    console.log(
-      `start getting merchantsInfos for ${JSON.stringify(merchantsIds[index])}`,
-    )
-    console.time('start getting merchantsInfos')
-    const merchant = await ifoodClient.getMerchantDetailsStatusAndInterruptions(
-      merchantsIds[index],
-    )
-    console.timeEnd('start getting merchantsInfos')
-    console.log(`catchau ${JSON.stringify(merchantsIds[index])}`)
-    merchants.push(merchant)
-  }
-  console.log(merchants.length)
-}
+import {
+  Merchant,
+  MerchantDetails,
+  MerchantInterruption,
+  MerchantOperation,
+  MerchantReviewInput,
+  MerchantReviewsInput,
+  MerchantSalesInput,
+} from './ifood/types/merchant'
+
+import AuthEventBus from './ifood/modules/auth/bus'
+import IfoodClientFinancial from './ifood/modules/financial'
+import IfoodClientMerchant from './ifood/modules/merchants'
+import Logger from './utils/logger'
+import IfoodClientOrder from './ifood/modules/order'
+import IfoodClientReview from './ifood/modules/reviews'
+import IfoodClientUtils from './ifood/utils'
+import { Sales } from './ifood/types/sales'
+import { Order, Review } from './ifood/types/reviews'
+export default class IfoodClient {
+  private logger = new Logger('ifood-client')
+
+  private authEventBus = AuthEventBus.getInstance()
+
+  private token = async () => await this.authEventBus.getToken()
```

Comment:
Faz mais sentido deixar a parte de variável de ambiente para a aplicação que está utilizando e passar via construtor ao inǘes de usar o AuthClient

Reply:
o token é gerado com o clientId e clientSecret é necessario fazer uma request para a api do ifood para gerar o token

Reply:
https://developer.ifood.com.br/pt-BR/docs/references/#operations-OAuth-oauthTokenV1
https://developer.ifood.com.br/pt-BR/docs/guides/authentication/

Reply:
Acho interessante, além de ter variável de ambiente, ter como passar via construtor da API. Por uma perspectiva da developer experience é melhor fazer as coisas mais explicitas, como o SDK da AWS faz: permite uso via variável de ambiente, mas também passando por parametro no construtor da instancia.